ckanext/unaids/tests/test_validators.py

```diff
@@ -27,3 +27,20 @@ def test_organisation_id_exists_validator(self):
                 {},
                 {}
             )
+
+    @pytest.mark.parametrize("format,resource_id,url,result", [
+        (None,  None, 'http://adr.local/resource/8919/demo.csv', 'text/csv'),
+        (None,  None, 'http://adr.local/resource/8919/demo.geojson', 'application/geo+json'),
+        (None,  None, 'http://adr.local/resource/8919/demo.pjnz', 'application/pjnz'),
+        ('text/csv',  '8919', 'http://adr.local/resource/89192/demo.xlsx', 'text/csv'),
+        (None,  '8919', 'http://adr.local/resource/8919/demo.csv', None)
+    ])
```

Comment:
I can't understand what exactly this test does. It'd be good to talk this over.

You can't rely on ckan web being up for running the tests. There's a test helper "resource with upload" which can create the test data you need. If it's supposed to be a fake URL it could be named "HTTP://fake.url/resource/1111/demo.csv"